core/src/main/java/io/jenkins/pluginhealth/scoring/probes/ProbeContext.java

```diff
@@ -88,6 +89,14 @@ public Optional<String> getRepositoryName(String scm) {
         return match.find() ? Optional.of(match.group("repo")) : Optional.empty();
     }
 
+    public void setIssueTrackerNameAndUrl(Map<String, String> issueTrackerNameAndUrl) {
+        this.issueTrackerNameAndUrl = issueTrackerNameAndUrl;
+    }
+
+    public Map<String, String> getIssueTrackerNameAndUrl() {
```

Comment:
Actually with `getIssueTrackerNameAndUrl` as a method name I would expect either to receive a `String` or just a `Tuple` or a custom object, but not a Map, so I guess using `issueTrackers` as a name would make more sense?

Reply:
I am curious, how do you expect to receive a particular datatype as a parameter based on getter and setter method name? You have said similar things in previous reviews as well. 

In the meeting, Adrien said that the name wasn't descriptive enough. And when he asked me what the method returns, I forgot too. Until I re-read my test cases. That is why I renamed them this way to remember that it returns `type` and `viewUrl` from IssueTrackers. 

The `ProbeContext` class has no JavaDocs for any of its members. Shall I add one in this case? Then I guess renaming won't be a problem.

Reply:
You are right, this sounds like preferences.

There are actually bunch of articles talking about good practices linked to naming in Java (or in languages in general). Considering code is much more read than written, it is important to optimize it for reading (code should be understandable easily and maintainable) rather than optimizing its writing.

In this particular context, what triggered me is the following: when I read `IssueTrackerNameAndUrl` I expect the method to return ONLY the issue tracker name and its url, but here the type that is returned is a `Map` which could contain more things, and which can actually contain several issue trackers. 

Naming is a very hard part of writing code, and it is quite a subjective thing, so it is normal to have discussions about that. IMHO, there are some global rules you can follow: plurals usually means you are talking about collections, etc.

For this particular variable, IMHO we could either go for something like `issueTrackers` but you are right that it might not be descriptive enough, so in that case I would usually go for something like `issueTrackerUrlsByNames` where I describe the content of the Map and how it is indexed.

Javadocs are also a good way to describe what is in variables, and give extra details, but that does not replace good naming (we should aim for both :stuck_out_tongue:)